aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/serializers/CauseSerializer.java

```diff
@@ -38,8 +44,9 @@ public CauseSerializer(JsonSerializer<Object> objectSerializer) {
     public void serialize(Cause cause, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
         if (!cause.getExceptions().isEmpty()) {
             ThrowableDescription first = cause.getExceptions().get(0);
-            if (null == first.getId() && null != first.getCause()) {
-                jsonGenerator.writeString(first.getCause());
+            String causeDescription = first.getCause();
```

Comment:
That would mean `addException` accepts `null` and populates it. Is that expected? Unfortunately nullness checker can only see that our code never passes `null` to `addException` but it wouldn't warn me if we allow it for users but I guess not.

Reply:
I wouldn't say it's expected, but I guess should be something we protect against since it is possible for customers to do. But that validation would belong in `EntityImpl` rather than over here. Probably not a big deal.

Reply:
Yeah agree that we'd want to do the checks in the `add` methods themselves, and it's probably not only cause. Let's do a deep scrub through everything in a separate PR.